File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
from collections import deque

def solution(progresses, speeds):
    
    n = len(progresses)
    deploys = deque()
    for idx in range(n):
        deploys.append(math.ceil((100-progresses[idx])/speeds[idx]))
    stack = []
    answer = []
    ##print(deploys)
    while deploys:
        
        work = deploys.popleft()
        ##print(stack,work)
        if stack and max_>=work:
            stack.append(work)
        else:
            answer.append(len(stack))
            stack=[work]
            max_=work

    if stack:
        answer.append(len(stack))
    return answer[1:]

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const works = []
    const N = speeds.length
    for(let i=0;i<N;i++){
        works.push(Math.ceil((100-progresses[i])/speeds[i]))
    }
    // console.log(works)
    let stack = [0]
    const answer = []
    let max_work = works[0]
    for(let i=1;i<N;i++){
        // console.log(stack)
        if(stack.length && max_work<works[i]){
            max_work = works[i]
           answer.push(stack.length)
            stack = []
        }
        stack.push(i)
        
    }
    if(stack.length){
        answer.push(stack.length)
    }
    
    return answer;
}